nf_core/pipeline-template/{{cookiecutter.name_noslash}}/modules/nf-core/fastqc.nf

```diff
@@ -10,14 +10,24 @@ process fastqc {
                 }
 
     input:
-    set val(name), file(reads) from ch_read_files_fastqc
+    tuple val(name), val(single_end), path(reads)
 
     output:
-    file "*_fastqc.{zip,html}" into ch_fastqc_results
+    path "*.{zip,html}"
 
     script:
-    """
-    fastqc --quiet --threads $task.cpus $reads
-    """
+    // Add soft-links to original FastQs for consistent naming in pipeline
+    if (single_end) {
+        """
+        [ ! -f  ${name}.fastq.gz ] && ln -s $reads ${name}.fastq.gz
+        fastqc --quiet --threads $task.cpus ${name}.fastq.gz
+        """
+    } else {
+        """
+        [ ! -f  ${name}_1.fastq.gz ] && ln -s ${reads[0]} ${name}_1.fastq.gz
+        [ ! -f  ${name}_2.fastq.gz ] && ln -s ${reads[1]} ${name}_2.fastq.gz
+        fastqc --quiet --threads $task.cpus ${name}_1.fastq.gz
+        fastqc --quiet --threads $task.cpus ${name}_2.fastq.gz
```

Comment:
Any reason for two separate commands?

```suggestion
        fastqc --quiet --threads $task.cpus ${name}_1.fastq.gz ${name}_2.fastq.gz
```

Reply:
Because I am soft-linking the original fastq files to those named by the sample names in the samplesheet. Cant use a glob otherwise it will run for the original and soft-linked fastq files.

Reply:
Ah, misunderstood. Good point 👍